harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -907,3 +946,73 @@ def amp_metrics_test(trial_class, training_metrics, agg_freq=1):
             if loss_prev is not None and metrics["stage"] == "one":
                 assert loss <= loss_prev, "loss was expected to decrease monotonically"
                 loss_prev = loss
+
+
+def create_trial_and_trial_controller(
```

Comment:
this looks like it's redoing some of what the Trainer does.

Can we test the Trainer codepath better, by using the Trainer (and just returning the (Trainer, TrialController, Trial) after fit() is done?

Reply:
discussed, decided to keep existing code because it tests internals, and also added new tests for Trainer path.